crates/bevy_winit/src/winit_windows.rs

```diff
@@ -25,6 +29,37 @@ impl WinitWindows {
         #[cfg(not(target_os = "windows"))]
         let mut winit_window_builder = winit::window::WindowBuilder::new();
 
+        let default_resize_constraints = WindowResizeConstraints::default();
+        let mut final_resize_constraints = window_descriptor.resize_constraints.clone();
+        if final_resize_constraints.min_width < default_resize_constraints.min_width {
+            final_resize_constraints.min_width = default_resize_constraints.min_width;
+        }
+        if final_resize_constraints.min_height < default_resize_constraints.min_height {
+            final_resize_constraints.min_height = default_resize_constraints.min_height;
+        }
+        if final_resize_constraints.max_width > default_resize_constraints.max_width {
+            final_resize_constraints.max_width = default_resize_constraints.max_width;
+        }
+        if final_resize_constraints.max_height > default_resize_constraints.max_height {
+            final_resize_constraints.max_height = default_resize_constraints.max_height;
+        }
+        if final_resize_constraints.max_width < final_resize_constraints.min_width {
+            final_resize_constraints.max_width = default_resize_constraints.max_width;
+        }
+        if final_resize_constraints.max_height < final_resize_constraints.min_height {
+            final_resize_constraints.max_height = default_resize_constraints.max_height;
+        }
```

Comment:
Ignoring the user's minimum size when it's less than the provided defaults is probably not the behaviour we wish to indroduce.

```suggestion
        let WindowResizeConstraints { mut min_width, mut min_height, mut max_width, mut max_height } = window_descriptor.resize_constraints.clone();
        min_width = min_width.max(1.);
        min_height = min_height.max(1.);
        if max_width < min_width {
            warn!("The given maximum width {} is smaller than the minimum width {}", max_width, min_width);
            max_width = min_width;
        }
        if max_height < min_height {
            warn!("The given maximum height {} is smaller than the minimum height {}", max_height, min_height);
            max_height = min_height;
        }
```